lib/toggles/constant_lookup.rb

```diff
@@ -8,35 +8,38 @@ def initialize(sym)
     end
   end
 
-
   # Return a tree walker that translates Module#const_missing(sym) into the next child node
   #
   # So Features::Cat::BearDog walks as:
   # * next_features = Feature.features # root
   # * const_missing(:Cat) => next_features = next_features['cat']
   # * const_missing(:BearDog) => next_features['bear_dog']
+  # * const_missing(:CN22) => next_features['c_n_22']
   #
   # Defined at Toggles.features_dir + "/cat/bear_dog.yaml"
   #
   # @raise [Error] if constant cannot be resolved
   def self.from(features, path)
     Class.new {
       class << self
-        attr_accessor :features
-        attr_accessor :path
+        attr_accessor :features, :path
 
         def const_missing(sym)
-          subtree_or_feature = features.fetch(
-            # translate class name into path part i.e :BearDog #=> 'bear_dog'
-            sym.to_s.gsub(/([a-z])([A-Z])/) { |s| s.chars.join('_') }.downcase.to_sym,
-          )
+          # translate class name into path part i.e :BearDog #=> 'bear_dog'
+          key = sym.to_s
+          key.gsub!(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2'.freeze)
+          key.gsub!(/([a-z\d])([A-Z])/, '\1_\2'.freeze)
+          key.downcase!
```

Comment:
I'm still seeing that we're having some issues with translation of potential keys, but let me know if you don't see it as an issue right now, or something we'll fix later.

```
[2] pry(main)> def tr(str)
[2] pry(main)*   str.gsub!(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2'.freeze)
[2] pry(main)*   str.gsub!(/([a-z\d])([A-Z])/, '\1_\2'.freeze)
[2] pry(main)*   str.downcase!
[2] pry(main)*   return str
[2] pry(main)* end
=> :tr
[3] pry(main)>
[4] pry(main)> tr("CN22")
=> "cn22"
[5] pry(main)> tr("Allow2DayAir")
=> "allow2_day_air"
[6] pry(main)> tr("Allow2ndDaySomething")
=> "allow2nd_day_something"
```

Reply:
As this a transitionary release and there is a workaround for it, I'm inclined to improve it within reason and then move on to a major release that removes support for module-based access entirely.

Reply:
cool, works for me